### ISSUE TITLE
persistent interface config on cisco devices

### ISSUE THREAD

**Issue by ku0100** [ktbyers/netmiko#614]
Hello Kirk,

Really enjoying the ease of netmiko over just using paramiko so thank you for creating this. I'm wondering if there is a way to keep an ssh session in a persistent interface config mode? To provide a little context, let's say I set up a connection and enter in config mode. If I want to configure a specific interface, it appears i need to submit a command like the below:

``config_commands = ["interface Gi1/1\ndescription this is a test"]``
``output = net_connect.send_config_set(config_commands)``

However, that work around is not very glamorous and somewhat inhibits what I'm ultimately trying to do (allow a user to dynamically enter commands as they would if they were actually on the router). Is there a method that you can provide on how I should tackle this or is the way I provided the only method?

Basically, I would like it so if the user is in config mode then they enter an interface, they are in that interface's config mode until they type exit.

Thanks again,

KW

**Comment by ktbyers** [comment]
@ku0100 FYI, you can send a list of commands:

```
config_commands = ["interface Gi1/1, description this is a test"]
output = net_connect.send_config_set(config_commands)
```

It will be in whatever context you put it in (in config mode). In other words, it will enter those commands one after the other in the context you are in. 

If you add the `exit_config_mode=False` argument, it won't exit config-mode (and should be in the context you left it in).

```
output = net_connect.send_config_set(config_commands, exit_config_mode=False)
```
